### ISSUE TITLE
Enhancement: Allow ensemble filters to include multiple base filters of the same type

### ISSUE THREAD

**Issue by annette987** [mlr-org/mlr#2688]
Currently, when creating an ensemble filter, the base filters must be unique. This is limiting for two reasons:
1. I cannot use both random forest minimal depth and random forest variable hunting in the same ensemble filter as they both have the same base filter -randomForestSRC_var.select
2. In this paper: http://www.cs.cornell.edu/~alexn/papers/shotgun.icml04.revised.rev2.pdf Caruana et al propose creating ensemble models from a large library of models, where there may be many models of the same base type, but with different parameters. Models are added to the ensemble one at a time using a greedy, forward step-wise selection process to add the model that improves the ensemble performance the most at each step. The same process could be applied to filter ensembles.

I propose this be achieved in the following way:

**Existing Method:**
```
ensemble = 
  makeFilterWrapper(
    makeLearner(cl="surv.coxph", id = "coxph", predict.type="response"), 
    fw.method = "E-mean",
    fw.base.methods=c("univariate.model.score", "mrmr"),
    fw.perc=0.3,
    more.args=list("univariate.model.score"=list(perf.learner=cox.lrn))
  )
```

**New Method:**
```
ensemble1 = 
  makeFilterWrapper(
    makeLearner(cl="surv.coxph", id = "coxph", predict.type="response"), 
    fw.method = "E-mean",
    fw.base.methods = list("filter1" = "univariate.model.score", "filter2"= "univariate.model.score"),
    fw.perc = 0.3,
    more.args=list("filter1"=list(perf.learner=cox.lrn), "filter2"=list(perf.learner=cox.lrn))
  )
```

1. The base methods are passed as a named list, where the names are identifiers and the values are the filter methods.
2. The arguments to the filters are passed in more.args as a named list, the same as in the current implementation, but in this case the names are the identifiers, not the filter names.
3. In generateFilterValues.R, we can no longer distinguish an ensemble method from a base method by assuming that only ensemble methods are passed as lists. So instead we check if the name of the filter exists in the FilterEnsembleRegister.

I have coded these and other changes neede and can do a PR if this idea is acceptable.

**Comment by larskotthoff** [comment]
That sounds good -- please go ahead and make the PR and we'll review it.

**Comment by pat-s** [comment]
fixed in #2693
